### PR TITLE
Adding an SSM Parameter Store Config source

### PR DESCRIPTION
**Pull request by zmarois**

### CODE REVIEW

File: priam/src/main/java/com/netflix/priam/SSMParameterStoreConfigSource.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    private final Map<String, String> data = Maps.newConcurrentMap();
+    private final ICredential provider;
+
+    @Inject
+    public SSMParameterStoreConfigSource(final ICredential provider) {
+        this.provider = provider;
+    }
+
+    @Override
+    public void intialize(final String appid, final String region) {
```
Comment (smagidson) [inline priam/src/main/java/com/netflix/priam/SSMParameterStoreConfigSource.java:59]: initialize* (need to also correct in the comment)

File: priam/src/main/java/com/netflix/priam/SSMParameterStoreConfigSource.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+import javax.inject.Inject;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Loads config data from SSM Paramter Store.  {@link #intialize(String, String)} will query Parameter store for any keys with prefix "PriamProperties-{apppId}-"
```
Comment (smagidson) [inline priam/src/main/java/com/netflix/priam/SSMParameterStoreConfigSource.java:41]: Parameter*

File: priam/src/main/java/com/netflix/priam/SSMParameterStoreConfigSource.java (hunk 1 above)
Comment (zmarois) [reply priam/src/main/java/com/netflix/priam/SSMParameterStoreConfigSource.java:59]: This is the interface's name for it. Gonna keep the spelling mistake to keep closer to base. Thanks though.

Comment (smagidson) [reply priam/src/main/java/com/netflix/priam/SSMParameterStoreConfigSource.java:59]: Oh dear. OK, makes sense.

**Comment by zmarois** [comment]
I'm reverting this all anyways. Gonna use default s3 encryption

**Pull request merged**